src/openrct2/drawing/Drawing.String.cpp

```diff
@@ -813,7 +819,30 @@ static const utf8* ttf_process_glyph_run(rct_drawpixelinfo* dpi, const utf8* tex
 
 static void ttf_process_string(rct_drawpixelinfo* dpi, const utf8* text, text_draw_info* info)
 {
-    const utf8* ch = text;
+    UErrorCode err = (UErrorCode)0;
+    // Force a hard left-to-right at the beginning (will mess up mixed strings' word order otherwise)
+    std::string text2 = std::string(u8"\xE2\x80\xAA") + text;
+
+    icu::UnicodeString ustr = icu::UnicodeString::fromUTF8(icu::StringPiece(text2));
+
+    int32_t length = ustr.length();
+    icu::UnicodeString reordered;
+    icu::UnicodeString shaped;
+    UBiDi* bidi = ubidi_openSized(length, 0, &err);
+    ubidi_setPara(bidi, ustr.getBuffer(), length, UBIDI_DEFAULT_LTR, nullptr, &err);
```

Comment:
`UBIDI_DEFAULT_LTR` is wrong for RTL languages.

Reply:
This PR is not finished. The language table will need an extra parameter to mark it as RTL anyway.
(Do not that this parameter has little effect, it's used if there are no explicit order marks.)